cli/lib/cmds/updateCache.js

```diff
@@ -19,52 +19,72 @@
 const fetch = require('node-fetch');
 const fs = require('fs');
 const UpdateCacheUrlProvider = require('amp-toolbox-update-cache');
+const errorRegex = /<br><br>\s+(.+?)\s+<ins>/;
+const MESSAGE_SUCCESS = '\x1b[32mSUCCESS!\x1b[0m';
+const MESSAGE_WARNING = '\x1b[33mWARNING!\x1b[0m';
 
 async function updateCaches(privateKey, url, logger) {
   logger.log(`Invalidating AMP Caches for ${url}`);
   try {
     const updateCacheUrlProvider = UpdateCacheUrlProvider.create(privateKey);
     const cacheUpdateUrls = await updateCacheUrlProvider.calculateFromOriginUrl(url);
-    cacheUpdateUrls.forEach(updateCache, logger);
-    return 0;
+    cacheUpdateUrls.forEach((cacheUpdateUrl) => updateCache(cacheUpdateUrl, logger));
   } catch (e) {
-    logger.error(`Error generating cache invalidation URL: ${e}`);
-    return 1;
+    throw new Error(`Error generating cache invalidation URL: ${e}`);
   }
 }
 
 async function updateCache(cacheUpdateUrlInfo, logger) {
+  const marker = `\x1b[1m[${cacheUpdateUrlInfo.cacheName}]:\x1b[0m`;
+  logger.log(`\n${marker} Invalidating ${cacheUpdateUrlInfo.cacheName}`);
+  logger.log(`${marker} Using Invalidation URL: ${cacheUpdateUrlInfo.updateCacheUrl}`);
+  let response;
   try {
-    logger.log(`\tInvalidating ${cacheUpdateUrlInfo.cacheName}`);
-    const response = await fetch(cacheUpdateUrlInfo.updateCacheUrl);
-    if (response.status !== 200) {
-      throw new Error(`Error Invalidating Cache URL: ${cacheUpdateUrlInfo.updateCacheUrl}`);
-    }
-    logger.log('\tSuccess!');
+    response = await fetch(cacheUpdateUrlInfo.updateCacheUrl);
   } catch (e) {
-    logger.error(`Error invalidating Cache URL: ${e}`);
+    logger.error(`[${cacheUpdateUrlInfo.cacheName}]: ${MESSAGE_WARNING} - Error connecting to ` +
+        `the AMP Cache, with message: "${e.message}"`);
   }
+
+  if (response.status !== 200) {
+    const body = await response.text();
+    const match = errorRegex.exec(body);
+
+    if (match) {
+      logger.error(
```

Comment:
these should also throw an error.

Reply:
This means the invalidation in *one* of the caches failed. Throwing an error would make the first failure cause the invalidation of all the following caches to fail. We could mark the message as an error, instead of warning.